src/geometry.rs

```diff
@@ -258,25 +286,47 @@ impl From<(u32, u32, u32)> for TriangleFace {
     }
 }
 
-/// A face edge. Contains indices to other geometry data - vertices
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub struct HalfEdge {
+/// Oriented face edge. Contains indices to other geometry data - vertices
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct OrientedEdge {
     pub vertices: (u32, u32),
 }
 
-impl HalfEdge {
+impl OrientedEdge {
     pub fn new(i1: u32, i2: u32) -> Self {
-        HalfEdge { vertices: (i1, i2) }
+        assert!(
+            i1 != i2,
+            "The oriented edge is constituted of the same vertex"
+        );
+        OrientedEdge { vertices: (i1, i2) }
+    }
+
+    pub fn is_reverted(self, other: OrientedEdge) -> bool {
+        self.vertices.0 == other.vertices.1 && self.vertices.1 == other.vertices.0
+    }
+}
+
+impl From<(u32, u32)> for OrientedEdge {
```

Comment:
Is this used? Let's not add all impls we can think of before they are used. They can be added later trivially.